src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/PEImage.cs

```diff
@@ -4,45 +4,35 @@
 
 using System;
 using System.Buffers;
-using System.Collections.Generic;
-using System.Collections.ObjectModel;
+using System.Collections.Immutable;
 using System.IO;
 using System.Linq;
+using System.Reflection.PortableExecutable;
 using System.Runtime.CompilerServices;
-using System.Text;
 
 namespace Microsoft.Diagnostics.Runtime.Utilities
 {
     /// <summary>
     /// A class to read information out of PE images (dll/exe).
     /// </summary>
-    public unsafe class PEImage
+    public class PEImage : IDisposable
     {
         private const ushort ExpectedDosHeaderMagic = 0x5A4D;   // MZ
         private const int PESignatureOffsetLocation = 0x3C;
         private const uint ExpectedPESignature = 0x00004550;    // PE00
-        private const int ImageDataDirectoryCount = 15;
-        private const int ComDataDirectory = 14;
-        private const int DebugDataDirectory = 6;
 
-        private readonly bool _virt;
+        private readonly PEReader _reader = null!;
```

Comment:
Maybe this should be a public property so others can directly use it if they want?

Reply:
Done.